docs/content/guides/developer/sui-101/sign-and-send-txn.mdx

```diff
@@ -0,0 +1,249 @@
+* The SDK's built-in gas estimation and coin selection picks the gas coin.
+* Sign the transaction to generate a [signature](/concepts/cryptography/transaction-auth/signatures#user-signature).
+* Submit the `TransactionBlock` and its signature for on-chain execution.
+
+:::info
+
+If you want to use a specific gas coin, first find the gas coin object ID to be used to pay for gas, and explicitly use that in the PTB. If there is no gas coin object, use the [splitCoin](./building-ptb#available-transactions) transaction to create a gas coin object. The split coin transaction should be the first transaction call in the PTB.
+
+:::
+
+## Examples
+
+The following examples demonstrate how to sign and execute transactions using Rust, TypeScript, or the Sui CLI.
+
+<Tabs groupId="code-language">
+
+<TabItem value="ts-sdk" label="TypeScript">
+
+There are various ways to instantiate a key pair and to derive its public key and Sui address using the Sui TypeScript SDK.
+
+```tsx
+    import { Ed25519Keypair } from '@mysten/sui.js/keypairs/ed25519';
+    import { Secp256k1Keypair } from '@mysten/sui.js/keypairs/secp256k1';
+    import { Secp256r1Keypair } from '@mysten/sui.js/keypairs/secp256r1';
+    import { fromHEX } from '@mysten/bcs';
+    import { SuiClient, getFullnodeUrl } from '@mysten/sui.js/client';
+    import { type Keypair } from '@mysten/sui.js/cryptography';
+    import { TransactionBlock } from '@mysten/sui.js/transactions';
+    
+    const kp_rand_0 = new Ed25519Keypair();
+    const kp_rand_1 = new Secp256k1Keypair();
+    const kp_rand_2 = new Secp256r1Keypair();
+
+    const kp_import_0 = Ed25519Keypair.fromSecretKey(fromHex("0xd463e11c7915945e86ac2b72d88b8190cfad8ff7b48e7eb892c275a5cf0a3e82"));
+    const kp_import_1 = Secp256k1Keypair.fromSecretKey(fromHex("0xd463e11c7915945e86ac2b72d88b8190cfad8ff7b48e7eb892c275a5cf0a3e82"));
+    const kp_import_2 = Secp256r1Keypair.fromSecretKey(fromHex("0xd463e11c7915945e86ac2b72d88b8190cfad8ff7b48e7eb892c275a5cf0a3e82"));
+
+	const kp_derive_0 = Ed25519Keypair.deriveKeypair("$MNEMONICS");
+	const kp_derive_1 = Secp256k1Keypair.deriveKeypair("$MNEMONICS");
+	const kp_derive_2 = Secp256r1Keypair.deriveKeypair("$MNEMONICS");
+
+	const kp_derive_with_path_0 = Ed25519Keypair.deriveKeypair("$MNEMONICS", "m/44'/784'/1'/0'/0'");
+	const kp_derive_with_path_1 = Secp256k1Keypair.deriveKeypair("$MNEMONICS", "m/54'/784'/1'/0/0");
+	const kp_derive_with_path_2 = Secp256r1Keypair.deriveKeypair("$MNEMONICS", "m/74'/784'/1'/0/0");
+
+    // replace `kp_rand_0` with the variable names above.
+    const pk = kp_rand_0.getPublicKey();
+    const sender = pk.toSuiAddress();
+
+    // create an example transaction block. 
+    const txb = new TransactionBlock();
+	txb.setSender(sender);
+	txb.setGasPrice(5);
+	txb.setGasBudget(100);
```

Comment:
is this in MIST?

Reply:
im trying to focus on the sign/execute tx part without going into the details here that can be distracting. this is just an example tx as mentioned, we discussed at the beginning that this for programmable txn building details refer to the other page.